runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -446,6 +481,13 @@ public synchronized CorfuRuntime connect() {
                 throw new RuntimeException(e);
             }
         }
+
+        if (loadSmrMapsAtConnect) {
+            FastSmrMapsLoader fastLoader = new FastSmrMapsLoader(this)
+                    .setBatchReadSize(bulkReadSizeForFastLoader)
+                    .setTimeoutInMinutesForLoading(timeoutInMinutesForFastLoading);
+            fastLoader.loadMaps();
+        }
```

Comment:
I agree with you. I was not sure if I was not missing something on my side. Since connect is synchronized, we should be safe.